### PR TITLE
Single result page for audio

### LINKED ISSUE #119: [Audio] Single result page for audio
Any first pass at this PR is welcome to write little to no styles, and instead focus on getting the needed information and components rendered. 

When styles _are_ implemented, they should be transitional:

- Please do not implement the _modal_ functionality yet; the contents of the modal can instead occupy the full page. 
- Stick to the accent blue and other temporary colors for now

> Reference mockup

CleanShot 2021-08-23 at 16 47 33@2x

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #119 by @zackkrida 

## Description

This PRs implements the Single Audio result page with transitional styles. It mainly organizes the page and components within it. The style changes for Audio Tracks, Tags, Tabs will be implemented in other PRs.

## Tests

Checkout the branch, run `npm run dev` and go to `localhost:8432/audio/123` to view the result.

## Screenshots

image

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/data/SampleAudioResponses.json [hunk 1]
```diff
@@ -642,39 +642,54 @@
...
+    ],
+    "audio_set": {
+      "name": "Distance & Temps",
+      "url": "https://www.jamendo.com/album/1290/distance-and-temps"
+    },
+    "duration": 248000,
+    "detail_url": "http://localhost:8000/v1/audio/242c1a4c-0623-436c-a600-7847412d28b6",
+    "related_url": "http://localhost:8000/v1/audio/242c1a4c-0623-436c-a600-7847412d28b6/recommendations"
   }
 }
```
Comment (obulat) [inline src/data/SampleAudioResponses.json:695]:
```suggestion
}

```

**Comment by zackkrida** [comment]
@obulat This is great, and I'm excited to see it refreshed once #207 lands. Some initial thoughts:

- Make the url path _singular_, `audio` instead of `audios`: http://localhost:8443/audios/8071128a-bebc-4c24-823d-6e26a2e6dfe2
- I know it's vague, because these are our transitional styles, but I think the page contents should be constrained to the header width here:

> before

CleanShot 2021-09-13 at 20 24 22@2x

> after
CleanShot 2021-09-13 at 20 25 57@2x
CleanShot 2021-09-13 at 20 26 00@2x


From a technical standpoint, the code looks great.

**Comment by obulat** [comment]
I've merged the updated AudioTrack component, here is the current result:

image

Note: when the audio object does not have duration property, the app doesn't start.

File: nuxt.config.js [hunk 2]
```diff
@@ -5,7 +5,10 @@ import locales from './src/locales/scripts/valid-locales.json'
  * Default environment variables are set on this key. Defaults are fallbacks to existing env vars.
  */
 export const env = {
-  apiUrl: process.env.API_URL || 'https://api.creativecommons.engineering/v1/',
+  apiUrl:
+    process.env.API_URL ||
+    // 'http://localhost:8000/v1/' ||
+    'https://api.creativecommons.engineering/v1/',
```
Comment (obulat) [inline nuxt.config.js:11]:
```suggestion
  apiUrl: process.env.API_URL ||  'https://api.creativecommons.engineering/v1/',
```

File: nuxt.config.js [hunk 3]
```diff
@@ -5,7 +5,7 @@ import locales from './src/locales/scripts/valid-locales.json'
  * Default environment variables are set on this key. Defaults are fallbacks to existing env vars.
  */
 export const env = {
-  apiUrl: process.env.API_URL || 'https://api.creativecommons.engineering/v1/',
+  apiUrl: process.env.API_URL ||  'https://api.creativecommons.engineering/v1/',
```
Comment (obulat) [inline nuxt.config.js:8]:
```suggestion
  apiUrl: process.env.API_URL || 'https://api.creativecommons.engineering/v1/',
```

**Review by krysal — APPROVED** [review]
This looks great! Don't you want to merge this before the store refactoring?

The only thing that seems odd is the text "There is no description available for most of the audios" but this if from Jamendo data. I think it would be better to show nothing for description instead.

**Review by zackkrida — APPROVED** [review]

**Comment by obulat** [comment]
> This looks great! Don't you want to merge this before the store refactoring?
> 
> The only thing that seems odd is the text "There is no description available for most of the audios" but this if from Jamendo data. I think it would be better to show nothing for description instead.

Thank you for mentioning that, @krysal ! I added it before talking to Francisco about the best solution for audios without description 😆  Will delete before merging!

**Pull request merged**